dev-packages/application-package/src/extension-package-collector.ts

```diff
@@ -52,17 +53,21 @@ export class ExtensionPackageCollector {
         this.parent = current;
     }
 
-    protected collectPackage(parentPackagePath: string, name: string, versionRange: string): void {
+    protected collectPackage(parentPackagePath: string, name: string, versionRange: string, optional: boolean): void {
         if (this.visited.has(name)) {
             return;
         }
-        this.visited.set(name, true);
+        this.visited.add(name);
 
         let packagePath: string | undefined;
         try {
             packagePath = this.resolveModule(parentPackagePath, name);
-        } catch {
-            console.debug(`Failed to resolve module: ${name}`);
+        } catch (err) {
+            if (optional) {
+                console.log(`Could not resolve optional peer dependency '${name}'. Skipping...`);
```

Comment:
Do we not still want the error message? Maybe the error is not that the package is missing (expected), but something unrelated?

Reply:
I'm not sure what the unrelated errors might be, since the only thing we actually do in the `try` block is to resolve the dependency to its `package.json` file. But I guess checking for `ENOENT` is an option.